From e71489b24a49899ca016068cbe39ad0576c52eb5 Mon Sep 17 00:00:00 2001
From: zhanchengbin <zhanchengbin1@huawei.com>
Date: Thu, 17 Mar 2022 10:53:18 +0800
Subject: [PATCH] tests: skip m_rootdir_acl if selinux is not disabled

Signed-off-by: zhanchengbin <zhanchengbin1@huawei.com>
---
 tests/m_rootdir_acl/script | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/tests/m_rootdir_acl/script b/tests/m_rootdir_acl/script
index e81c82c..6b78d3d 100644
--- a/tests/m_rootdir_acl/script
+++ b/tests/m_rootdir_acl/script
@@ -16,6 +16,12 @@ if [ "$os" = "GNU" ]; then
 	return 0
 fi
 
+selinux=`grep -nRE "^SELINUX=" /etc/selinux/config | awk -F "=" '{print $2}'`
+if [ $selinux != "disabled" ]; then
+	echo "$test_name: $test_description: skipped (selinux is not disabled)"
+	return 0
+fi
+
 MKFS_DIR=$TMPFILE.dir
 OUT=$test_name.log
 EXP=$test_dir/expect
-- 
1.8.3.1

